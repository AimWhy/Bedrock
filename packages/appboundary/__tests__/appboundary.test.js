// import React from 'react';
// import { create } from 'react-test-renderer';
// import { spacing } from '@bedrock-layout/spacing-constants';
// import { ThemeProvider } from 'styled-components';
import AppBoundary from '../src';

// const Lorem = () => (
//   <>
//     {Array.from(Array(4).keys()).map(i => (
//       <p key={i}>
//         Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in
//         vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus
//         neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum
//         sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien,
//         condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant
//         morbi tristique senectus et netus et malesuada fames ac turpis egestas.
//         Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper
//         diam, ac placerat purus tortor et enim.
//       </p>
//     ))}
//   </>
// );

describe('AppBoundary', () => {
  describe('correct usage', () => {
    test('AppBoundary is not null', () => {
      expect(AppBoundary).toBeTruthy();
    });

    // it('renders default gutters', () => {
    //   const appboundary = create(
    //     <AppBoundary>
    //       <Lorem />
    //     </AppBoundary>
    //   );
    //   expect(appboundary.toJSON()).toMatchSnapshot();
    // });

    // it('renders all the gutter options', () => {
    //   Object.keys(spacing).forEach(gutter => {
    //     const appboundary = create(
    //       <AppBoundary gutter={gutter}>
    //         <Lorem />
    //       </AppBoundary>
    //     );
    //     expect(appboundary.toJSON()).toMatchSnapshot();
    //   });
    // });

    // it('renders custom minItemWidth', () => {
    //   const appboundary = create(
    //     <AppBoundary minItemWidth={320}>
    //       <Lorem />
    //     </AppBoundary>
    //   );
    //   expect(appboundary.toJSON()).toMatchSnapshot();
    // });

    // it('renders with theme overrides', () => {
    //   const appboundary = create(
    //     <ThemeProvider theme={{ spacing: { md: '200px' } }}>
    //       <AppBoundary>
    //         <Lorem />
    //       </AppBoundary>
    //     </ThemeProvider>
    //   );
    //   expect(appboundary.toJSON()).toMatchSnapshot();
    // });
  });

  // describe('incorrect usage', () => {
  //   let originalError;
  //   let spy;
  //   beforeEach(() => {
  //     originalError = console.error;
  //     spy = jest.fn();
  //     console.error = spy;
  //   });
  //   afterEach(() => {
  //     console.error = originalError;
  //   });

  //   it('renders default with console error with wrong gutter input', () => {
  //     expect(spy.mock.calls.length).toBe(0);

  //     const errorStack = create(
  //       <AppBoundary gutter='incorrect'>
  //         <Lorem />
  //       </AppBoundary>
  //     );

  //     expect(spy.mock.calls.length).toBe(1);
  //     expect(errorStack.toJSON()).toMatchSnapshot();
  //   });

  //   it('renders default with console error with minItemWidth input', () => {
  //     expect(spy.mock.calls.length).toBe(0);

  //     const errorStack = create(
  //       <AppBoundary minItemWidth='incorrect'>
  //         <Lorem />
  //       </AppBoundary>
  //     );

  //     expect(spy.mock.calls.length).toBe(1);
  //     expect(errorStack.toJSON()).toMatchSnapshot();
  //   });
  // });
});
