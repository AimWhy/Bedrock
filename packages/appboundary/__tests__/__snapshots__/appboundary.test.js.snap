// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBoundary correct usage renders default gutters 1`] = `
.c1 {
  --maxWidth: 1920px;
  box-sizing: content-box;
  margin-inline-start: auto;
  margin-inline-end: auto;
  margin-inline: auto;
  max-inline-size: 1023px;
}

.c0 {
  padding: 0;
  max-width: 100%;
  overflow: hidden;
  height: 100%;
}

@supports (max-inline-size:1920px) {
  .c1 {
    max-inline-size: var(--maxWidth,1023px);
  }
}

<div
  className="c0"
  data-bedrock-layout-appboundary=""
>
  <div
    className="c1"
    data-bedrock-layout-center=""
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus tortor et enim.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus tortor et enim.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus tortor et enim.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus tortor et enim.
    </p>
  </div>
</div>
`;

exports[`AppBoundary correct usage renders with theme overrides 1`] = `
.c1 {
  --maxWidth: 1920px;
  box-sizing: content-box;
  margin-inline-start: auto;
  margin-inline-end: auto;
  margin-inline: auto;
  max-inline-size: 1023px;
}

.c0 {
  padding: 0;
  max-width: 100%;
  overflow: hidden;
  height: 100%;
}

@supports (max-inline-size:1920px) {
  .c1 {
    max-inline-size: var(--maxWidth,1023px);
  }
}

<div
  className="c0"
  data-bedrock-layout-appboundary=""
>
  <div
    className="c1"
    data-bedrock-layout-center=""
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus tortor et enim.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus tortor et enim.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus tortor et enim.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus tortor et enim.
    </p>
  </div>
</div>
`;
